source/MaterialXRuntime/RtStage.cpp

```diff
@@ -141,11 +141,30 @@ void RtStage::restorePrim(const RtPath& parentPath, const RtPrim& prim)
     _cast(_ptr)->restorePrim(*static_cast<PvtPath*>(parentPath._ptr), prim);
 }
 
+RtPrim RtStage::getImplementation(const RtNodeDef& definition) const
+{
+    const RtToken& nodeDefName = definition.getName();
+    const RtToken& nodeDefVersion = definition.getVersion();
+
+    RtSchemaPredicate<RtNodeGraph> filter;
+    for (RtPrim child : _cast(_ptr)->getRootPrim()->getChildren(filter))
+    {
+        RtNodeGraph nodeGraph(child);
+        if (nodeGraph.getDefinition() == nodeDefName &&
+            nodeGraph.getVersion() == nodeDefVersion)
+        {
+            PvtPrim* graphPrim = PvtObject::ptr<PvtPrim>(child);
+            return RtPrim(graphPrim->hnd());
+        }
+    }
+    return RtPrim();
```

Comment:
Is this or throwing an exception better?

Reply:
Not sure. Currently definitions which are not nodegraphs have no support added yet in which case it should not throw.